### ISSUE TITLE
Program demands interaction with invisible subwindow

### ISSUE THREAD

**Issue by CodeF53** [ATLauncher/ATLauncher#715]
**Describe the bug**
When installing mods ATLauncher will sometimes create the mod installation popup behind the rest of the windows, leaving it inaccessible.

Once this occurs, the following is true:
- error noise when you click any ATLauncher window
- the only way to close ATLauncher is by opening task manager and killing java or with the tray icon

**To Reproduce**
I have not been able to find a consistent way to make the issue occur
- Install a lot of mods all at once?
- Something to do with dependencies?

**Expected behavior**
The window that needs attention should always be visible.

**Video of problem happening**

https://user-images.githubusercontent.com/37855219/213902232-c44bdbb9-cc90-404d-9c30-e9fedee6125f.mp4

**System information:**
- Windows 11

**Additional context**
Its really hard to get this to consistently happen.

Hopefully this issue can get enough people sharing their experience so we can find a way to do it consistently

**Comment by CodeF53** [comment]
https://user-images.githubusercontent.com/37855219/213902418-e9949138-ce40-4133-aef5-29d6070a0b16.mp4

Installing a mod twice?

**Comment by RyanTheAllmighty** [comment]
Yeah it happens randomly in various places for the past couple years. It used to be much worse, but was mostly fixed, but there's some places it comes up an not sure why, I'm expecting it to be a Swing/Java bug honestly, but could be just parenting not being set correctly on windows and dialogs

**Comment by CodeF53** [comment]
I don't quite get how this is low priority, it happens very often for me when setting up a new instance, and is one of the most infuriating bugs I have ever experienced.
Having to restart the launcher in the middle of setting up is just nails on a chalkboard annoying

**Comment by Obscure2020** [comment]
I agree with @CodeF53. I wouldn't consider this low-priority. Even if it doesn't seem to happen often _**TO YOU,**_ it's still a pretty fun-stopping event when it does occur. Requiring the user to kill and reopen the entire program is not a viable way to allow the program to continue operating.

Now, I've tried replicating @CodeF53's specific failure cases on my own machine without success. I _have_ personally experienced this bug before, but for me it seemed to stem from leaving ATLauncher in the background with a subwindow open for too long. I'm still not 100% sure. If I ever figure out a consistent, reproducible method, I'll share it later.

For now, I'd like to suggest a new workaround for you to implement, @RyanTheAllmighty. There are a few things that seem to remain consistent when this bug appears:
1. ATLauncher begins behaving like there is some sort of invisible subwindow open, and it demands that you interact with this invisible subwindow.
2. The taskbar thingy (what is that tiny icon area called?) still works.

Stemming from these, I propose a new entry in the right-click list for that tiny icon (besides `Toggle Console`, `Open Launcher Folder`, and `Quit`): a new option called `Kill Subwindows`. If possible, clicking this option would force all subwindows to close _but keep the main window and the overall program running,_ providing a faster way to recover from this glitch.

**Comment by Obscure2020** [comment]
Oh, and one addendum: `Kill Subwindows` should kill all windows other than the main window _and the console._ The main window should remain open, and the console should also remain open _if_ it was already open.

**Comment by RyanTheAllmighty** [comment]
I've added in a tray menu to kill open dialogs which should work around the issue for now. I'll leave the issue open until the underlying issue can be fixed, but everything looks good as I expected in terms of windows being properly parented, still very likely to believe it's a Swing/Java issue. Unless it can be reliably replicated, it's unlikely to get fixed

**Comment by Obscure2020** [comment]
Thanks for implementing the workaround so fast! I'll be sure to let you know if we find a reliable way to replicate the problem.

**Comment by sirsnowy7** [comment]
This happens for me, strange this is the workaround doesn't work, and it always happens 100% if I click the next page button. Linux, newest ver. of app

**Comment by RyanTheAllmighty** [comment]
Yeah it's unfortunately random and difficult to reproduce for myself which makes it a pain to try and fix sorry

**Comment by seriousfreezing** [comment]
It's happened to me so many times that I hate this bug

**Comment by Madis0** [comment]
Could consistently reproduce it twice right after installing a Modrinth modpack.

**Comment by Sorro123** [comment]
Not fixed, still happens to me on the latest version, can barely install 2 mods before having to reset it

**Comment by seriousfreezing** [comment]
It also happens to me from time to time, but at least there's a way to kill the suspended windows

**Comment by RyanTheAllmighty** [comment]
Reopening this as it's clear it's not fixed.

I have found a way to reproduce fairly reliably and that's by going to Packs -> Modrinth -> then install on Fabulously Optimized (probably any of them, but that was on the top) and then close the dialog. Sometimes it happens on the first go, sometimes it takes 10-20 times, but it does eventually happen

**Comment by RyanTheAllmighty** [comment]
Looking into this and one thing I've noticed is that it happens when we fetch the versions for a modpack from Modrinth:

https://github.com/ATLauncher/ATLauncher/blob/2c604a75d0b7f41e1518dd5d961f7299daea135e/src/main/java/com/atlauncher/gui/dialogs/InstanceInstallerDialog.java#L637-L641

I've isolated it to this specific call. I've also observed that it happens pretty quick with a pack with more versions, but when you switch to sorting by newest packs and therefor likely to only have 1 version available, the issue doesn't occur.

Makes me think there's some memory issue loading all those versions that then escalates to the issues with the window getting locked up.

**Comment by RyanTheAllmighty** [comment]
The issue seems to be with the caching:

https://github.com/ATLauncher/ATLauncher/blob/2c604a75d0b7f41e1518dd5d961f7299daea135e/src/main/java/com/atlauncher/utils/ModrinthApi.java#L292

Removing the cached call on this api call and the issue goes away.

**Comment by RyanTheAllmighty** [comment]
I've spent the better part of a day now digging through profilers and trying to find out why this is happening. I cannot for the life of me understand where the issue is coming from.

Since the issue only occurs when caching that versions Modrinth endpoint, I will remove caching from that call and any others that I find exhibit the same behaviour.

**Comment by RyanTheAllmighty** [comment]
Alright I have tried another fix with version 3.4.41.3 if anyone who was having consistent issues can verify if it works or not that'd be great
